### PR TITLE
All semver comparisons now ensure it's safe to do so

### PR DESCRIPTION
**Pull request by RubenRBS**
Did not run prettier nor the linter to avoid a hard-to-read diff, but will do so before merging after I've ensured every change is ok

### CODE REVIEW

**Comment by RubenRBS** [comment]
Some Zig semver checks seem to be outdated. Will fix on a different PR once this gets merged

File: lib/utils.ts [hunk 1]
```diff
@@ -412,3 +410,24 @@ export function countOccurrences<T>(collection: Iterable<T>, item: T): number {
     }
     return result;
 }
+
+export function asSafeVer(semver: string | number | null) {
+    if (semver != null) {
```
Comment (mattgodbolt) [inline lib/utils.ts:415]: Does this need to be `!==` or is there some subtle JS thing going on here?

Comment (RubenRBS) [reply lib/utils.ts:415]: I don't think undefined _can_ be passed right now, so you're right, it should be strong eq

**Comment by RubenRBS** [comment]
Test on staging looked good. Will push the formatting changes now and merge

**Pull request merged**